index.js

```diff
@@ -116,16 +116,15 @@ exports.errorLogger = function errorLogger(options) {
 
     options.requestWhitelist = options.requestWhitelist || exports.requestWhitelist;
     options.requestFilter = options.requestFilter || exports.defaultRequestFilter;
-    options.winstonInstance = options.winstonInstance || (winston.createLogger({
-      transports: options.transports,
-      format: options.format
-    }));
+    const getLogger = loggerFactory(options);
     options.msg = options.msg || 'middlewareError';
     options.baseMeta = options.baseMeta || {};
     options.metaField = options.metaField || null;
     options.level = options.level || 'error';
     options.dynamicMeta = options.dynamicMeta || function(req, res, err) { return null; };
-    const exceptionHandler = new winston.ExceptionHandler(options.winstonInstance);
+    // TODO: improve the default exception meta information by not calling a 
+    // constructor than a function that does nothing with the constructor parameter.
+    const exceptionHandler = new winston.ExceptionHandler({});
```

Comment:
can you explain a bit more why this change? `ExceptionHandler` needs the logger instance

Reply:
The logger instance is not known at application startup. Which is when this line is executing. The logger instance is known at execution time through the 'getLogger' method.

The only reason why there is an ExceptionHandler is to call getAllInfo on it. 

I can think of 2 ways to solve this problem.

1. Build an exception handler on each requests, passing the logger instance. In other works, call these lines (or something similar) on all requests: 
`options.exceptionToMeta = options.exceptionToMeta || exceptionHandler.getAllInfo.bind(exceptionHandler);
`

2. Accept and embrace the fact that ExceptionHandler.getAllInfo does not require the logger instance to build the exception metadata. You can view the implementation here: https://github.com/winstonjs/winston/blob/8423c03c59c31f61f57519b4097420a464c90f83/lib/winston/exception-handler.js#L75

I chose option 2 because it seemed a reasonable tradeoff between readability and a small iterative change.

Reply:
@golopot want to throw your two cents here?

Reply:
Line 119 - 122 seems to provide evidence that this can be called with `this.options.winstonInstance` initialized. I don't think it is good for performance to create a new instance every time errorLogger is created. I think a better solution would be to leave this as it was and find a good place to remove `this.options.winstonInstance` so that a new instance can be created here only if you need it.